### PR TITLE
identity/group: fix bug with inconsistent plan on upgrade

### LINKED ISSUE #1690: Regression in 3.10.0: Error: Provider produced inconsistent final plan
During a plan/apply run in TFE, an error was produced:

> Error: Provider produced inconsistent final plan
> When expanding the plan for vault_identity_group.group["RG-Foo-Bar-Baz"] to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/vault" produced an invalid new value for .external_member_group_ids: was cty.False, but now null.
> 
> This is a bug in the provider, which should be reported in the provider's own issue tracker.

external_member_group_ids is a new resource attribute added in provider 3.10.0.

The plan (transcribing from the TFE GUI) showed:

```
+ external_member_group_ids: false
~ member_group_ids: [
    each existing value -> Known after apply
  ]
```

The apply then produced the error text shown above.

The code change that triggered this plan/apply run changed other members of the `for_each` for the `vault_identity_group.group` resource block, but made no changes to the particular resource instances for which the error arose.

### Terraform Version
Terraform 1.3.4
terraform-provider-vault 3.10.0

### Affected Resource(s)
- vault_identity_group

### Terraform code
```hcl
resource "vault_identity_group" "group" {
  for_each = <complex expression trimmed>
  name = each.key
  type = "internal"
  member_group_ids = <complex expression trimmed>
  policies = concat(each.value, ["default"])
  depends_on = [var.vault-namespace]
}
```

### PR DESCRIPTION
**Pull request by fairclothjm**
### Description

Good description of the problem here: https://github.com/hashicorp/terraform-provider-vault/issues/1690#issuecomment-1382953551

Closes #1690

### Checklist
- [x] Added [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md) entry (only for user-facing changes)
- [ ] Acceptance tests where run against all supported Vault Versions

### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

File: vault/resource_identity_group.go [hunk 1]
```diff
@@ -128,11 +136,13 @@ func identityGroupUpdateFields(d *schema.ResourceData, data map[string]interface
...
 		// Member groups and entities can't be set for external groups
 		if d.Get("type").(string) == "internal" {
-			if externalMemberEntityIds, ok := d.GetOk("external_member_entity_ids"); !(ok && externalMemberEntityIds.(bool)) {
+			externalMemberEntityIds := d.Get("external_member_entity_ids")
+			if !externalMemberEntityIds.(bool) {
 				data["member_entity_ids"] = d.Get("member_entity_ids").(*schema.Set).List()
 			}
 
-			if externalMemberGroupIds, ok := d.GetOk("external_member_group_ids"); !(ok && externalMemberGroupIds.(bool)) {
+			externalMemberGroupIds := d.Get("external_member_group_ids")
```
Comment (fairclothjm) [inline vault/resource_identity_group.go:144]: This wasn't required to fix the inconsistent plan error but we recommend using `d.Get` for booleans  :)

**Comment by fairclothjm** [comment]
I am not exactly sure how to test this since it requires different versions of the provider as far as I know.

File: vault/resource_identity_group.go (hunk 1 above)
Comment (vinay-gopalan) [reply vault/resource_identity_group.go:144]: Nice catch! Thanks for fixing

**Comment by vinay-gopalan** [comment]
Going to close this in favor of #2043 after discussion 🙏🏼

**Pull request closed without merging**